ydb/core/blobstorage/vdisk/balance/handoff_map.h

```diff
@@ -202,117 +174,110 @@ namespace NKikimr {
+        Y_DEFER { Counter++; };
+
         if (!keepData) {
-            result = TrRes.SetRmData(key, memRec, dataMerger);
-        } else {
-            result = TrRes.SetRaw(key, memRec, dataMerger); // unchanged by default
-            if (RunHandoff) {
-                Y_DEBUG_ABORT_UNLESS(MoveMap.size() == DelMap.size() && Counter < MoveMap.size());
-                const NMatrix::TVectorType localVec = memRec->GetIngress().LocalParts(Top->GType);
-                ui8 vecSize = Top->GType.TotalPartCount();
-                const NMatrix::TVectorType movePlan(MoveMap.at(Counter), vecSize);
-                const NMatrix::TVectorType delPlan(DelMap.at(Counter), vecSize);
-
-                // handle move
-                const NMatrix::TVectorType moveVec = movePlan & localVec;
-                if (!moveVec.Empty()) {
-                    if (memRec->GetType() == TBlobType::HugeBlob) {
-                        Y_ABORT("Implement"); // FIXME
-                    } else if (memRec->GetType() == TBlobType::DiskBlob) {
-                        const auto& merger = dataMerger->GetDiskBlobMerger();
-                        const TDiskBlob& blob = merger.GetDiskBlob();
-                        ui32 fullDataSize = blob.GetFullDataSize();
-
-                        // iterate via parts
-                        for (TDiskBlob::TPartIterator it = blob.begin(), e = blob.end(); it != e; ++it) {
-                            ui8 partId = it.GetPartId();
-                            Y_DEBUG_ABORT_UNLESS(partId > 0);
-                            if (moveVec.Get(partId - 1)) {
-                                TLogoBlobID id(key.LogoBlobID(), partId);
-                                TVDiskIdShort vdisk(memRec->GetIngress().GetMainReplica(Top.get(), id));
-                                // FIXME: for huge blobs we need to implement an actor which makes read, because we don't have
-                                // this blob in memory
-                                TRope temp;
-                                bool res = HandoffDelegate->Restore(ctx, vdisk, id, fullDataSize, it.GetPart());
-                                Stat.SuccessfulMoveSend += ui64(res);
-                                Stat.FailedMoveSend += ui64(!res);
-                            }
-                        }
-                    } else
-                        Y_ABORT("Unexpected case");
+            return TrRes.SetRmData(key, memRec, dataMerger);
+        }
+
+        const TTransformedItem *defaultResult = TrRes.SetRaw(key, memRec, dataMerger); // unchanged by default
+        if (!RunHandoff) {
+            return defaultResult;
+        }
+
+        Y_DEBUG_ABORT_UNLESS(Counter < DelMap.size());
+        const NMatrix::TVectorType localVec = memRec->GetIngress().LocalParts(Top->GType);
+        ui8 vecSize = Top->GType.TotalPartCount();
+
+        const NMatrix::TVectorType delPlan(DelMap.at(Counter), vecSize);
+        const NMatrix::TVectorType delVec = delPlan & localVec;
```

Comment:
what's the meaning of this? DelMap already contains filtered (by local vector) parts, does it?

Reply:
DelMap is constructed in BuildMap, it iterates and merge memRecs over whole lsm, not the specific level compacting now. That the reason why we could have bit set in delPlan and don't have certain bit in localVec